base/commands/queue/queue_poll.go

```diff
@@ -0,0 +1,103 @@
+//go:build base || queue
+
+package _queue
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hazelcast/hazelcast-commandline-client/clc"
+	. "github.com/hazelcast/hazelcast-commandline-client/internal/check"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/output"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/plug"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/proto/codec"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/serialization"
+	"github.com/hazelcast/hazelcast-go-client"
+)
+
+const flagCount = "count"
+
+type QueuePollCommand struct {
+}
+
+func (qc *QueuePollCommand) Init(cc plug.InitContext) error {
+	addValueTypeFlag(cc)
+	help := "Remove the given number of elements from the given Queue"
+	cc.AddIntFlag(flagCount, "", 1, false, "number of element to be removed from the given queue")
+	cc.SetCommandHelp(help, help)
+	cc.SetCommandUsage("poll [flags]")
+	cc.SetPositionalArgCount(0, 0)
+	return nil
+}
+
+func (qc *QueuePollCommand) Exec(ctx context.Context, ec plug.ExecContext) error {
+	queueName := ec.Props().GetString(queueFlagName)
+	count := int(ec.Props().GetInt(flagCount))
+	ci, err := ec.ClientInternal(ctx)
+	if err != nil {
+		return err
+	}
+	var rows []output.Row
+	for i := 0; i < count; i++ {
+		valueType, value, err := qc.poll(ctx, ec, ci, queueName, err)
+		if err != nil {
+			return err
+		}
+		row := output.Row{
+			output.Column{
+				Name:  output.NameValue,
+				Type:  valueType,
+				Value: value,
+			},
+		}
+		if ec.Props().GetBool(queueFlagShowType) {
+			row = append(row, output.Column{
+				Name:  output.NameValueType,
+				Type:  serialization.TypeString,
+				Value: serialization.TypeToLabel(valueType),
+			})
+		}
+		rows = append(rows, row)
+	}
```

Comment:
I think this code should be refactored to run in in an ExecuteBlocking block, instead of running each poll in its own ExecuteBlocking block.